var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -577,6 +587,10 @@ def edit(self, spec, prefix):
                     "-DOFFLOAD_TARGET=cuda",
                 ]
                 libs += ["-lcublas"]
+
+                if spec.satisfies("+cuda_fft"):
+                    cppflags += ["-D__PW_CUDA"]
+                    libs += ["-lcufft"]
```

Comment:
you disabled the availability of the `cuda_fft` variant above I think

Reply:
From the perspective of Spack recipe directives, this branch is entered only for version `9.*`. Not sure if this makes sense for `cp2k` though.

Reply:
@alalazo exactly what I meant. In this PR there is also a `when="@:9",  # req in CP2K v2022+` added for the `cuda_fft` variant.

Reply:
I do not think `cuda_fft` is disabled above as it is enabled by default in `@2022:`. That is why I set the condition on the variant. However, looking at the cp2k docs again, I see that it _can_ be disabled in the 2022 version. If I am understanding correctly, versions up to 9 would use `D__PW_CUDA` to turn it on, and versions 2022 and above would use `D__NO_OFFLOAD_PW` to turn it off. So, I should redo the logic to reflect that. Does that sound correct @dev-zero?

Reply:
I added a commit to enable the `cuda_fft` variant for `@2022:` and adjusted the `if...` logic.

Reply:
I think that looks right, yes.
Thanks a lot!